website/content/docs/platform/k8s/vso/sources/vault/gcp-auth.mdx

```diff
@@ -0,0 +1,80 @@
+---
+layout: docs
+page_title: GCP auth support for Vault Secrets Operator
+description: >-
+  Learn how GCP authentication works for Vault Secrets Operator
+---
+
+# GCP auth support for Vault Secrets Operator
+
+The Vault Secrets Operator (VSO) supports authenticating to Vault's [GCP auth](/vault/docs/auth/gcp) method, using Google's Kubernetes Engine (GKE) workload identity.
+
+1. Follow Google's [Use Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) to enable workload identity on a GKE cluster so your Kubernetes service account can impersonate a Google IAM service account. 
+
+1. Create an appropriate authentication role in your Vault instance:
+
+  <CodeTabs>
+  <CodeBlockConfig>
+
+  ```shell-session
+  $ vault write auth/gcp/role/<VAULT_GCP_ROLE> \
+      type="iam" \
+      policies="default" \
+      max_jwt_exp=3600 \
+      bound_service_accounts="<SERVICE_ACCOUNT>@<GCP_PROJECT>.iam.gserviceaccount.com"
+  ```
+
+  </CodeBlockConfig>
+  <CodeBlockConfig>
+
+  ```hcl
+  resource "vault_gcp_auth_backend_role" "gcp_role" {
+    backend                = "auth/gcp"
+    role                   = <VAULT_GCP_ROLE>
+    type                   = "iam"
+    token_policies         = "default"
+    max_jwt_exp            = 3600
+    bound_service_accounts = [
+      "<SERVICE_ACCOUNT>@<GCP_PROJECT>.iam.gserviceaccount.com",
+    ]
+  }
+  ```
+
+  </CodeBlockConfig>
+  </CodeTabs>
+
+  <Note>
+
+    `max_jwt_exp` needs to be greater than or equal to 1 hour (3600)
+  
+  </Note>
+
+1. Create the corresponding authentication object for VSO:
+
+  ```yaml
+  apiVersion: secrets.hashicorp.com/v1beta1
+  kind: VaultAuth
+  metadata:
+    name: vaultauth-gcp-example
+    namespace: <K8S_NAMESPACE>
+  spec:
+    vaultConnectionRef: <VAULT_CONNECTION_NAME>
+    mount: gcp
+    method: gcp
+    gcp:
+      role: <VAULT_GCP_ROLE>
+      workloadIdentityServiceAccount: <SERVICE_ACCOUNT>
```

Comment:
Is it required to be in exactly this form, or do we also support it with the `@<GCP_PROJECT>.iam.gserviceaccount.com` suffix like it appears above? Reading the page with fresh eyes, it's not totally obvious that the suffix was intentionally omitted. I kind of wish we had the ability to interactively substitute in real values like Google's docs can.

Reply:
Oh good question, this is the k8s service account, which often is just be the `<SERVICE_ACCOUNT>` portion of the GCP service account, but I'll change this to `<K8S_SERVICE_ACCOUNT>` to make it more clear which is expected here.

And yeah, +1 to Google's doc substitution, it's really nice for things like this.